lib/wallet/wallet.js

```diff
@@ -1820,14 +1824,77 @@ class Wallet extends EventEmitter {
     }
   }
 
+  /**
+   * Create and finalize a bid & a reveal (in advance)
+   * MTX with a lock.
+   * @param {String} name
+   * @param {Number} value
+   * @param {Number} lockup
+   * @param {Object} options
+   */
+
+  async createAuctionTxs(name, value, lockup, options) {
+    const unlock = await this.fundLock.lock();
+    try {
+      return await this._createAuctionTxs(name, value, lockup, options);
+    } finally {
+      unlock();
+    }
+  }
+
+  /**
+   * Create and finalize a bid & a reveal (in advance)
+   * MTX without a lock.
+   * @param {String} name
+   * @param {Number} value
+   * @param {Number} lockup
+   * @param {Object} options
+   */
+
+  async _createAuctionTxs(name, value, lockup, options) {
+    const bid = await this._createBid(name, value, lockup, options);
+
+    // Prepare the data needed to make the reveal in advance
+    const rawName = Buffer.from(name, 'ascii');
+    const nameHash = rules.hashName(rawName);
+    const bidBlind = new BlindBid();
+    bidBlind.name = name;
+    bidBlind.nameHash = nameHash;
+    bidBlind.prevout = Outpoint.fromTX(bid, 0);
+    bidBlind.value = bid;
+    bidBlind.lockup = lockup;
+    bidBlind.blind = bid.outputs[0].covenant.getHash(3);
+    bidBlind.own = true;
```

Comment:
Even though the BID transaction has not been broadcast yet and it's coins are not yet known to the wallet, at this point in time the `bidBlind` should actually already be saved and available in the `txdb`. It is generated inside `makeBid()` which is called by `_createBid()` up on line 1855:

https://github.com/handshake-org/hsd/blob/4042141f1aa28a887179fd0ddbf78489b053dedf/lib/wallet/wallet.js#L1739

What I think might work here is, after you `_createBid()`, loop through the outputs of that MTX and find the actual BID covenant (it could be at any index). When you have the BID covenant you can pull the blind value from it and then:

```js
      const blind = coin.covenant.getHash(3);
      const bv = await this.getBlind(blind);
```

Grabbing the output of the BID TX will also enable you to get the coin you want to pass to `mtx.fund()` (see other comment)

In fact, it may make sense to just leave the existing `makeReveal()` function alone entirely, but copy the output construction logic we need to this new function.

I think if this approach works, it'll mean you don't need to do the extra module export stuff from TXDB that you got above.

Reply:
The thought behind altering reveal is to keep linked code in the same place. Sure it add complexity to the function, but it limits the odds of a function to be "forgotten" when introducing changes. Will think about it